packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1884,6 +1885,90 @@ function commitPassiveMount(
   }
 }
 
+function invokeEffectMountInDEV(fiber: Fiber, hookFlags: HookFlags): void {
+  if (__DEV__ && enableDoubleInvokingEffects) {
+    switch (fiber.tag) {
+      case FunctionComponent:
+      case ForwardRef:
+      case SimpleMemoComponent:
+      case Block: {
+        if ((fiber.flags & hookFlags) !== NoFlags) {
```

Comment:
Technically this check is redundant. `invokeEffectsInDev` (in the work loop) already checks `fiber.flags` before calling this function. That's why we don't have to check if `instance.componentDidMount` is defined below before calling it.

I think a similar analogy would be `commitResetTextContent`. `commitMutationEffectsImpl` (in the work loop) checks `flags & ContentReset` before calling `commitResetTextContent` so it doesn't re-check.